pkgs/games/openttd/jgrpp.nix

```diff
@@ -10,6 +10,7 @@ openttd.overrideAttrs (oldAttrs: rec {
     rev = "jgrpp-${version}";
     hash = "sha256-aEmuwWNxqjuLmiWMrqwTUtCwdwDMA00eQspCHntxEXw=";
   };
+  patches = [];
```

Comment:
Musing a bit about this -- this will stomp all existing patches injected by other overrides on `openttd`. If a user had an overlay and wanted to apply a patch in `openttd` and then expected those patches to make it through to `openttd-jgrpp`. 

Is that someone that someone would ever do though? 

I'm okay with it, because I would probably just override `openttd-jrpp` again..

Reply:
I would say that ideally the packages should not reuse one another's implementation using `overrideAttrs` unless they are almost identical. In this case even source trees used in `openttd` and `jgrpp` are different. This breakage was quite a surprise to me.

Reply:
Agreed. 

I think a better pattern is having a generic openttd builder function with `src`, `patches` and `version` arguments. 

A good example of this is https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/ms/msgpack/package.nix